calculator.js

```diff
@@ -0,0 +1,150 @@
+/* global math */
+// Variables
+var buttons = [['7','8','9','+'],['4','5','6','-'],['1','2','3','*'],['0','.','=','/']];
+var theme = 'dark';
+var lastNum = '=';
+
+function prep() {
+  // Create base elements for the page
+  var container = document.createElement('div');
+  var header = document.createElement('div');
+  var nav = document.createElement('div');
+  var main = document.createElement('div');
+  var aside = document.createElement('div');
+  var footer = document.createElement('div');
+
+  // Apply IDs to the base elements
+  container.id = 'container';
+  header.id = 'header';
+  nav.id = 'nav';
+  main.id = 'main';
+  aside.id = 'aside';
+  footer.id = 'footer';
+
+  // Append main elements to container
+  container.appendChild(header);
+  container.appendChild(nav);
+  container.appendChild(main);
+  container.appendChild(aside);
+  container.appendChild(footer);
+
+  // Add container to page
+  document.body.appendChild(container);
+
+  heading(header);
```

Comment:
Btw, I like this approach of passing the parent container to the function that constructs the elements. I think it's a slightly better approach, though, to have the functions return the element, and then call `appendChild` outside the function.

So instead it would look like:

```js
header.appendChild(heading());
main.appendChild(calculator());
footer.appendChild(themeButtons());
```

This keeps the `heading`, `calculator`, and `themeButtons` functions "pure" (no side-effects).